src/Symfony/Component/DependencyInjection/ParameterBag/ParameterResolver.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\ParameterBag;
+
+use Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException;
+use Symfony\Component\DependencyInjection\Exception\ParameterCircularReferenceException;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+final class ParameterResolver implements ParameterResolverInterface
```

Comment:
What's the problem with non-generic name if you type-hint against the interface everywhere?

Also, if you think that different implementation is very unlikely, why do you need an interface? I mean, callable allows to change resolving logic, the rest of resolver logic is very generic as you said.